app/models/taxon.rb

```diff
@@ -20,12 +20,13 @@ def tagged_content
     @tagged_content ||= fetch_tagged_content
   end
 
-  def services_content
-    @services_content ||= fetch_most_popular_content('services')
-  end
-
-  def guidance_and_regulation_content
-    @guidance_and_regulation_content ||= fetch_most_popular_content('guidance_and_regulation')
+  def section_content(supergroup)
```

Comment:
Given that this section-for-supergroup logic has been moved to the `Taxon` class you could remove all specifics of the supergroup names by passing it directly to the `fetch_most_popular_content` method...


```ruby
def section_content(supergroup)
  section_contents[supergroup]
end

def section_contents
  @section_contents ||= {}
end

def fetch_most_popular_content(content_purpose_supergroup)
  @section_contents[supergroup] ||= MostPopularContent.fetch(content_id: content_id, filter_content_purpose_supergroup: content_purpose_supergroup)
end
```

Reply:
@steventux I think we wanted to keep them separate for now, as other sections (e.g: 'news and communication') will not use the `fetch_most_popular_content` method as they need to be ordered differently.

Reply:
sorry @vanitabarrett initial comment wasn't entirely coherent - I've updated the code snippet accordingly

Reply:
Ah I see :+1: makes sense.